app/models/concerns/account_settings.rb

```diff
@@ -192,33 +192,36 @@ def reload_library_config
     end
 
     def reload_analytics
+      # rubocop:disable Style/RedundantSelf
       # fall back to the default values if they aren't set in the tenant
-      unless google_analytics_id.present? &&
-             google_oauth_app_name.present? &&
-             google_oauth_app_version.present? &&
-             google_oauth_private_key_secret.present? &&
-             google_oauth_client_email.present? &&
-             (google_oauth_private_key_value.present? || google_oauth_private_key_path.present?)
+      unless self.google_analytics_id.present? &&
+             self.google_oauth_app_name.present? &&
+             self.google_oauth_app_version.present? &&
+             (self.google_oauth_private_key_value.present? || self.google_oauth_private_key_path.present?) &&
+             self.google_oauth_private_key_secret.present? &&
+             self.google_oauth_client_email.present?
 
         config = Hyrax::Analytics::Config.load_from_yaml
-        google_analytics_id ||= config.analytics_id
-        google_oauth_app_name ||= config.app_name
-        google_oauth_app_version ||= config.app_version
-        google_oauth_private_key_secret ||= config.privkey_secret
-        google_oauth_private_key_value ||= config.privkey_value
-        google_oauth_client_email ||= config.client_email
+        self.google_analytics_id = self.google_analytics_id.presence || config.analytics_id
+        self.google_oauth_app_name = self.google_oauth_app_name.presence || config.app_name
+        self.google_oauth_app_version = self.google_oauth_app_version.presence || config.app_version
+        self.google_oauth_private_key_value = self.google_oauth_private_key_value.presence || config.privkey_value
+        self.google_oauth_private_key_path = self.google_oauth_private_key_path.presence || config.privkey_path
+        self.google_oauth_private_key_secret = self.google_oauth_private_key_secret.presence || config.privkey_secret
+        self.google_oauth_client_email = self.google_oauth_client_email.presence || config.client_email
```

Comment:
guard against the account setting value being an empty string.

if an account once had analytics settings that have been removed, the value for each would be an empty string. when using the or/equal operator with an empty string and non empty string, the empty string is returned. we cannot however check for the presence of a variable while also using the or/equal operator.